app/views/BookDetail/index.tsx

```diff
@@ -91,10 +127,13 @@ function SimilarBook(props: BookProps) {
         </div>
     );
 }
-function BookDetail(props: BookProps) {
-    const {
-        book,
-    } = props;
+function BookDetail() {
+    const { id } = useParams();
+
+    const { data: result, loading, error } = useQuery(BOOK_DETAIL, {
+        variables: { id },
+    });
+    const book = (!loading && result) ? result.book : [];
```

Comment:
```suggestion
    const book = result?.book ?? [];
```

Also, should be default value be an empty list?